### ISSUE TITLE
Is there anyone find that the train_shape.ipynb has very bad segmentation result for square? 

### ISSUE THREAD

**Issue by easycui** [matterport/Mask_RCNN#269]
As shown in following image:
![train_shapes](https://user-images.githubusercontent.com/941132/36492851-22abffb2-16fc-11e8-81a0-55aedbcde880.png)
The predictions of triangle and circle are reasonable.  Have tried to train with more epochs, but it doesn't help at all. Even predicting in the training set, the prediction of squares is as bad as the test set.

**Comment by alwansm** [comment]
yes. not only for squares. I was wondering if the parameters or the size of the training set should be changed.
Did you find any explanation for that?

**Comment by Ghostlzw** [comment]
I had the same problems before, but I improved the accuracy by increasing the value of STEPS_PER_EPOCH to 500 and changed the value of VALIDATION_STEPS (I am not sure whether this variable would work) to 25. Certainly, it takes more time to train the model after increasing the values.
Finally I got this result. 
![image](https://user-images.githubusercontent.com/15846067/36978602-e337580e-20bf-11e8-9715-b369f83a23c7.png)

**Comment by alwansm** [comment]
Hello @Ghostlzw  thanks for the clarification. did you try to test your own image and not on random image from th dataset? so instead of 'original_image'  use 'image' where image=cv2.imread(YOUR IMAGE)

**Comment by Ghostlzw** [comment]
Sorry @alwansm , I am also a beginner of the code. And I'm trying to understand the whole code and apply the code to my own dataset. So if you have any idea how to train your own dataset and detect them, please let me know. Thanks.

**Comment by alwansm** [comment]
In reality, I have the same issue. there is no clear explanation of how to train our own datasets. Basically, I drew my own circles and squares using paint  (save it as an image) and the trained model was not able to detect them correctly (not accurate)!

**Comment by rida1896** [comment]
Where you able to train on your own dataset?

**Comment by waleedka** [comment]
@easycui You're right about the squares, there was a bug that caused masks that cover the entire mask area (like in solid rectangles) to be reset to zero, and that was visible in the case of squares. That bug was fixed a while ago. 

Regarding the other discussion about training on your own dataset, see this post that explains how to train on your own dataset from scratch:
https://engineering.matterport.com/splash-of-color-instance-segmentation-with-mask-r-cnn-and-tensorflow-7c761e238b46
